docs/design/gluster-block-provisioning.md

```diff
@@ -0,0 +1,350 @@
+# Gluster-Block Provisioning End-To-End Design
+
+## Motivation
+
+While GlusterFS volumes can be used for RWO as well as RWX (and ROX) volumes,
+for some workloads, specifically metadata-heavy, the RWO support is not
+satisfactory with respect to performance. Furthermore, the scalability
+in number of volumes is currently not very good.
+
```

Comment:
@obnoxxx Simply put, all RWO providers are expected to enforce locking semantics to a host. You are correct in that the AccessMode in k8s is purely descriptive. It tells the developer what semantics they are getting. Taking a RWX volume and describing it as RWO in the PV description will cause developers to submit claims against an RWO volume but instead get back a Shared FileSystem with no locking capabilities. So they think they are protected from another user in their same namespace using the claim in a different pod/container, but really they are not. It is not about performance, it is about locking to a host. Existing CNS volumes do not do this, thus they are not RWO.